server/src/main/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregator.java

```diff
@@ -277,226 +228,13 @@ public void collectDebugInfo(BiConsumer<String, Object> add) {
         add.accept("filters", filtersDebug);
     }
 
-    /**
-     * Collects results by running each filter against the searcher and doesn't
-     * build any {@link LeafBucketCollector}s which is generally faster than
-     * {@link Compatible} but doesn't support when there is a parent aggregator
-     * or any child aggregators.
-     */
-    public static class FilterByFilter extends FiltersAggregator {
```

Comment:
Most of this class is just moved into its own file. The cost estimation is entirely gone.